### ISSUE TITLE
npm rebuild fails because of missing deps/cld/binding.gyp

### ISSUE THREAD

**Issue by kylehg** [dachev/node-cld#20]
Although this package builds just fine on first install, running `npm rebuild` results in an error because `deps` has been deleted in the post-install step.

```
kylehardgrave@ObvsMBP65:test $ npm rebuild

> cld@2.4.3 install /Users/kylehardgrave/test/node_modules/cld
> node-gyp rebuild

gyp: deps/cld/binding.gyp not found (cwd: /Users/kylehardgrave/test/node_modules/cld) while loading dependencies of binding.gyp while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/opt/boxen/nodenv/versions/v0.12.2/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:343:16)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Darwin 14.3.0
gyp ERR! command "node" "/opt/boxen/nodenv/versions/v0.12.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/kylehardgrave/test/node_modules/cld
gyp ERR! node -v v0.12.2
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok 
```

Note this is on Node v0.12.2, OS X Yosemite 10.10.3, cld v2.4.3.

Full `npm-debug.log` below:

```
0 info it worked if it ends with ok
1 verbose cli [ 'node',
1 verbose cli   '/opt/boxen/nodenv/versions/v0.12.2/bin/npm',
1 verbose cli   'rebuild' ]
2 info using npm@2.7.6
3 info using node@v0.12.2
4 info readInstalled object
5 verbose rebuild path, id [ '/Users/kylehardgrave/test', 'test@1.0.0' ]
6 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld', 'cld@2.4.3' ]
7 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob',
7 verbose rebuild   'glob@5.0.13' ]
8 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/inflight',
8 verbose rebuild   'inflight@1.0.4' ]
9 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/once',
9 verbose rebuild   'once@1.3.2' ]
10 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/once/node_modules/wrappy',
10 verbose rebuild   'wrappy@1.0.1' ]
11 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/inflight/node_modules/wrappy',
11 verbose rebuild   'wrappy@1.0.1' ]
12 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/inherits',
12 verbose rebuild   'inherits@2.0.1' ]
13 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/minimatch',
13 verbose rebuild   'minimatch@2.0.8' ]
14 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion',
14 verbose rebuild   'brace-expansion@1.1.0' ]
15 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion/node_modules/balanced-match',
15 verbose rebuild   'balanced-match@0.2.0' ]
16 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion/node_modules/concat-map',
16 verbose rebuild   'concat-map@0.0.1' ]
17 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/path-is-absolute',
17 verbose rebuild   'path-is-absolute@1.0.0' ]
18 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/nan',
18 verbose rebuild   'nan@1.8.4' ]
19 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf',
19 verbose rebuild   'rimraf@2.4.1' ]
20 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob',
20 verbose rebuild   'glob@4.5.3' ]
21 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/inflight',
21 verbose rebuild   'inflight@1.0.4' ]
22 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/once',
22 verbose rebuild   'once@1.3.2' ]
23 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/once/node_modules/wrappy',
23 verbose rebuild   'wrappy@1.0.1' ]
24 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/inflight/node_modules/wrappy',
24 verbose rebuild   'wrappy@1.0.1' ]
25 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/inherits',
25 verbose rebuild   'inherits@2.0.1' ]
26 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/minimatch',
26 verbose rebuild   'minimatch@2.0.8' ]
27 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion',
27 verbose rebuild   'brace-expansion@1.1.0' ]
28 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion/node_modules/balanced-match',
28 verbose rebuild   'balanced-match@0.2.0' ]
29 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion/node_modules/concat-map',
29 verbose rebuild   'concat-map@0.0.1' ]
30 verbose rebuild path, id [ '/Users/kylehardgrave/test/node_modules/cld/node_modules/underscore',
30 verbose rebuild   'underscore@1.8.3' ]
31 silly rebuild set [ '/Users/kylehardgrave/test/node_modules/cld',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/inflight',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/once',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/once/node_modules/wrappy',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/inflight/node_modules/wrappy',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/inherits',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/minimatch',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion/node_modules/balanced-match',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion/node_modules/concat-map',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/glob/node_modules/path-is-absolute',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/nan',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/inflight',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/once',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/once/node_modules/wrappy',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/inflight/node_modules/wrappy',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/inherits',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/minimatch',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion/node_modules/balanced-match',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/rimraf/node_modules/glob/node_modules/minimatch/node_modules/brace-expansion/node_modules/concat-map',
31 silly rebuild set   '/Users/kylehardgrave/test/node_modules/cld/node_modules/underscore' ]
32 info build /Users/kylehardgrave/test/node_modules/cld
33 info preinstall cld@2.4.3
34 info linkStuff cld@2.4.3
35 silly linkStuff cld@2.4.3 has /Users/kylehardgrave/test/node_modules as its parent node_modules
36 verbose linkBins cld@2.4.3
37 verbose linkMans cld@2.4.3
38 verbose rebuildBundles cld@2.4.3
39 verbose rebuildBundles [ '.bin', 'glob', 'nan', 'rimraf', 'underscore' ]
40 info install cld@2.4.3
41 verbose unsafe-perm in lifecycle true
42 info cld@2.4.3 Failed to exec install script
43 verbose stack Error: cld@2.4.3 install: `node-gyp rebuild`
43 verbose stack Exit status 1
43 verbose stack     at EventEmitter.<anonymous> (/opt/boxen/nodenv/versions/v0.12.2/lib/node_modules/npm/lib/utils/lifecycle.js:213:16)
43 verbose stack     at EventEmitter.emit (events.js:110:17)
43 verbose stack     at ChildProcess.<anonymous> (/opt/boxen/nodenv/versions/v0.12.2/lib/node_modules/npm/lib/utils/spawn.js:14:12)
43 verbose stack     at ChildProcess.emit (events.js:110:17)
43 verbose stack     at maybeClose (child_process.js:1015:16)
43 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
44 verbose pkgid cld@2.4.3
45 verbose cwd /Users/kylehardgrave/test
46 error Darwin 14.3.0
47 error argv "node" "/opt/boxen/nodenv/versions/v0.12.2/bin/npm" "rebuild"
48 error node v0.12.2
49 error npm  v2.7.6
50 error code ELIFECYCLE
51 error cld@2.4.3 install: `node-gyp rebuild`
51 error Exit status 1
52 error Failed at the cld@2.4.3 install script 'node-gyp rebuild'.
52 error This is most likely a problem with the cld package,
52 error not with npm itself.
52 error Tell the author that this fails on your system:
52 error     node-gyp rebuild
52 error You can get their info via:
52 error     npm owner ls cld
52 error There is likely additional logging output above.
53 verbose exit [ 1, true ]
```

**Comment by dachev** [comment]
Hi @kylehg. The assumption behind deleting ```deps``` was that ```rebuild``` will never be triggered in the normal lifecycle of an npm-managed application. How did you run into this?

**Comment by kylehg** [comment]
Hey @dachev! Our node application checks in all our `node_modules` to source control in order to ensure reproducible builds, but the caveat to this is we must run `npm rebuild` to compile any C code for the appropriate platform. Being able to check in node modules like this is an important requirement for us to running Node in production. 

Does that make sense?

**Comment by dachev** [comment]
I had a feeling this was a bad idea but ran with it anyway. I'll revert this change and publish a new build in the next fe days.

**Comment by kylehg** [comment]
Awesome, thank @dachev! Idk if you saw but I already submitted a PR (#21) that fixes the issue, if it'd be easier to just use that

**Comment by dachev** [comment]
Ah, I didn't realize it was you. Thanks.

**Comment by dachev** [comment]
Published 2.4.4

**Comment by kylehg** [comment]
Thanks @dachev!
